components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/internal/IdentityMgtComponent.java

```diff
@@ -241,8 +242,9 @@ protected void registerCachingService(CarbonCachingService cachingService, Map<S
 //    }
 
     protected void unregisterCachingService(CarbonCachingService carbonCachingService) {
-
-        IdentityMgtDataHolder.getInstance().registerCacheService(null);
+        if (this.carbonCachingService == carbonCachingService) {
```

Comment:
Is there a special reason to have this equals check? Any harm in simply executing null assignment?